--- a/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
+++ b/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
@@ -1,7 +1,7 @@
 #define RWNX_HE_CAPABILITIES                                    \
 {                                                               \
-    .has_he = false,                                            \
+    .has_he = true,                                             \
     .he_cap_elem = {                                            \
         .mac_capab = {                                          \
             .ht40 = true,                                       \
