--- a/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
+++ b/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
@@ -4997,7 +4997,7 @@
  * @change_beacon: Change the beacon parameters for an access point mode
  *	interface. This should reject the call when AP mode wasn't started.
  */
-#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0)
+#if 1
 static int rwnx_cfg80211_change_beacon(struct wiphy *wiphy, struct net_device *dev,
 				       struct cfg80211_ap_update *info)
 #else
@@ -5017,7 +5017,7 @@
 	elem.dma_addr = 0;
 
 	// Build the beacon
-#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0)
+#if 1
 	buf = rwnx_build_bcn(bcn, &info->beacon);
 #else
 	buf = rwnx_build_bcn(bcn, info);
@@ -5114,7 +5114,7 @@
 }
 #endif
 
-static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
+static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy, struct net_device *net_device,
 					     struct cfg80211_chan_def *chandef)
 {
 	struct rwnx_hw *rwnx_hw = wiphy_priv(wiphy);
@@ -5611,7 +5611,7 @@
 	if (rwnx_vif->vif_index == rwnx_hw->monitor_vif){
 		//retrieve channel from firmware
-        rwnx_cfg80211_set_monitor_channel(wiphy, NULL);
+        rwnx_cfg80211_set_monitor_channel(wiphy, 0, NULL);
 	}
 
 	//Check if channel context is valid
--- a/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c
+++ b/src/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c
@@ -1396,7 +1396,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
 			    &ctxt->chan_def,
 #endif
-                       NL80211_RADAR_CAC_FINISHED, GFP_KERNEL);
+                       NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
 
 	rwnx_send_apm_stop_cac_req(rwnx_hw, radar->cac_vif);
 	rwnx_chanctx_unlink(radar->cac_vif);
@@ -1496,7 +1496,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
 			    &ctxt->chan_def,
 #endif
-                           NL80211_RADAR_CAC_ABORTED, GFP_KERNEL);
+                           NL80211_RADAR_CAC_ABORTED, GFP_KERNEL, 0);
 
 	rwnx_chanctx_unlink(radar->cac_vif);
 }
